### DISCUSSION TITLE
Support for third-party packages

### DISCUSSION THREAD

**Discussion by ollieread** [sprout-laravel/sprout#72] [Ideas]
Sprout is built to work with Laravel, which means it will work with everything that comes with `laravel/laravel`. It is extremely rare, however, that a project would remain using only this, and not include any third-party packages.

With that in mind, I'd like to hear from you about which packages you think Sprout should support, or you'd like Sprout to support. I have my internal plans with a rough idea of the packages that I'd like to look into supporting, but really, it's all about what you the users, need.

If you've got a package, or multiple, that you'd like to see Sprout natively support, respond to this discussion with your list! If you see someone already suggesting one or more, give it a vote!

Likewise, if you're the developer of a third-party package, and you'd like to see Sprout support your package, or you'd like to support Sprout, please let me know as well!

> [!NOTE]
> Package suggestions can be anything that currently works with Laravel, whether they're Laravel packages themselves, or external third-party packages.

**Comment by alancolant** [comment]
Ideally, support for:  
- **Laravel Telescope**  
- **Laravel Horizon** (at least the ability to inject the tenant context into a queued job, even in native Laravel).  
- **Laravel Octane** (I use it a lot personally, but I'm not sure how popular it really is).  

For external packages:  
- **FilamentPHP** (and by extension, Livewire, since it depends on it).

**Reply by ollieread** [reply]
>     * **Laravel Telescope**

This is an interesting one, as where people may want it within Telescope would depend on what they're building. That being said, having it tacked onto data in the same way that authenticated users are would be my initial thinking.

>     * **Laravel Horizon** (at least the ability to inject the tenant context into a queued job, even in native Laravel).

The `JobOverride` class provides this. It uses the `JobProcessing` event to set the current tenant based on the values within Laravels' context.

>     * **Laravel Octane** (I use it a lot personally, but I'm not sure how popular it really is).

This is definitely going to be an interesting one, though I think we've made some good progress on this.

>     * **FilamentPHP** (and by extension, Livewire, since it depends on it).

This too will be interesting, but again, we've made some good progress in that direction!

**Reply by alancolant** [reply]
Of course, I’ve added what we already discussed as well.

Alright for the jobs, that should ensure Horizon runs smoothly.
For "telescope," the tag-based approach would already be a good start.

**Comment by andreas-freitag** [comment]
Without knowing too much details about sprout, these packages are the ones that could be potentially difficult (even though I think they should probably work out of the box):

- spatie/laravel-activitylog
- spatie/laravel-backup (for multi-db version)
- spatie/laravel-medialibrary
- spatie/laravel-permission

**Reply by ollieread** [reply]
Thanks! I'll add them to the list

**Reply by ollieread** [reply]
I had a bit of a look into it, and it looks like this shouldn't be too complex.

> spatie/laravel-activitylog

This package lets you provide a custom model, so that works for both multi and single database.

> spatie/laravel-backup (for multi-db version)

For multi-db I'd need to override the scoped binding for `Config` to include the tenant databases.

> spatie/laravel-medialibrary

Also supports custom models.

> spatie/laravel-permission

Supports custom models too.

**Comment by ollieread** [comment]
I've put together a task to look into these and collect information.

https://github.com/orgs/sprout-laravel/projects/3/views/2?query=sort:updated-desc%20is:open&pane=issue&itemId=87847812
